### ISSUE TITLE
Query Loop block displays incorrect posts when inheriting template query and filtering multiple terms per taxonomy

### ISSUE THREAD

**Issue by robrobsn** [WordPress/gutenberg#35019]
### Description

The Query Loop block doesn't display the correct list of posts when inheriting query from template and filtering multiple terms per taxonomy ( e.g.: ?category_name=block,media ). It **only** shows posts assigned to the first term. Using the same url and switching to a classic theme ( e.g.: TwentyTwentyOne ) the correct list of posts is displayed.

### Step-by-step reproduction instructions

1. Activate a FSE theme ( e.g.: Blockbase )
2. Add taxonomy filter parameters to your WordPress URL - multiple terms for one taxonomy (e.g: `?category_name=block,media` or `/category/block,media/`). Make sure there are posts that are assigned to **each and only one term** of the chosen terms ( e.g.: Post `A` has **only** category `block` and Post `B` has **only** category `media` ).
3. Observe the list of posts being displayed. E.g: **Only Post** `A` is being displayed
4. Flip the terms in the URL ( e.g.: `/category/media,block/` )
5. Observe the list of posts being displayed. E.g: **Only Post** `B` is being displayed
6. Copy the URL and activate a classic non-FSE theme ( e.g.: TwentyTwentyOne )
7. Reload the copied URL ( e.g.: `/category/media,block/`  )
8. Observe the list of posts being displayed. E.g: **Post** `A` **and Post** `B` are being displayed

### Screenshots, screen recording, code snippet

The issue is caused by the new local `WP_Query` and the way the `$query_args` array is created when inheriting the query in the function `gutenberg_render_block_core_post_template` ( plugins/gutenberg/build/block-library/blocks/post-template.php:17 ). The `$query_args` array is based on the global `$wp_query->query_vars`. Those query vars have been parsed and now have **only one** taxonomy term set when filtering for multiple ones (e.g.: `/category/block,media/` results in `$query_vars['category_name']=> 'block']` ) and thus the new query filters only by one category. 

This also creates issues with pagination. The pagination is based on the correct amount of posts which can result in an empty last page.

The Query Loop works as expected if the `$query_args` are based on the global `$wp_query->query` which still holds all terms.
Adding `$query_args = $wp_query->query;` At the end of the `if ( $use_global_query ) {` block on line 35 of `plugins/gutenberg/build/block-library/blocks/post-template.php`. 

```php
function gutenberg_render_block_core_post_template( $attributes, $content, $block ) {
	...
	if ( $use_global_query ) {
		global $wp_query;
		...
		$query_args = $wp_query->query;
	}
```
Adding this line of code fixes the pagination as well. 
The code comments mention the need to unset the `offset` param which would have to be refactored if still necessary. 

Hope that helps, either as a fix or at least a start! 

### Environment info

WordPress 5.8.1
Gutenberg 11.5.1

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by torounit** [comment]
Tested in my local environment.

WP 6.0 reproduces this problem, but it seems to be fixed in Gutenberg 13.8.0-rc2.

Maybe it was fixed in https://github.com/WordPress/gutenberg/pull/40656.

**Comment by ndiego** [comment]
I tested as well using Gutenberg 13.7.3 and the issue appears resolved. Therefore I am going to close this issue.
